### PR TITLE
Story/tp65873 Search left panel - round 4 (requirements)

### LINKED ISSUE #3: Animation during screen updates
One of the great things about an SPA is that the entire page stays constant, while new data is added/removed/updated.

The full screen animation during update delays really defeats that. 

(I recognize I'm looking at a very early version...but the point of writing these issues is to focus on getting some early "tuning".

### PR DESCRIPTION
**Pull request by billdami**
Closes #3, #24

### CODE REVIEW

File: pep/app/pods/components/search/refine/component.ts [hunk 1]
```diff
@@ -30,15 +31,18 @@ interface SearchRefineArgs {
...
+    @service configuration!: ConfigurationService;
 
     @tracked expandedGroups: string[] = [];
 
     get groups() {
+        const cfg = this.configuration.base.search;
         const groups: RefineGroup[] = [];
         const fieldsMap = this.args.metadata?.facetCounts?.facet_fields ?? {};
         const incFields = Object.keys(fieldsMap);
+        const displayedFacets = cfg.facets.defaultFields.map((id) => SEARCH_FACETS.findBy('id', id)!);
```
Comment (bakerac4) [inline pep/app/pods/components/search/refine/component.ts:43]: I have a feeling using `!` is probably the wrong way to go here

File: pep/app/pods/most-cited/template.hbs [hunk 2]
```diff
@@ -9,12 +9,12 @@
                 }}
             </h4>
             <div class="d-flex align-items-center">
-                <Button
+                <Button::WithTooltip
                     @type="link"
                     @icon="download"
                     @iconSize="lg"
                     @action={{this.downloadCSV}}
-                    title={{t "common.export"}}
```
Comment (bakerac4) [inline pep/app/pods/most-cited/template.hbs:17]: How does this work with accessibility? Do you still need title as well?

File: pep/app/pods/most-viewed/template.hbs [hunk 3]
```diff
@@ -9,12 +9,12 @@
...
             </h4>
             <div class="d-flex align-items-center">
-                <Button
+                <Button::WithTooltip
                     @type="link"
                     @icon="download"
                     @iconSize="lg"
                     @action={{this.downloadCSV}}
-                    title={{t "common.export"}}
+                    @tooltip={{t "common.export"}}
```
Comment (bakerac4) [inline pep/app/pods/most-viewed/template.hbs:17]: Same thing here, does this change accessibility?

File: pep/app/pods/read/document/controller.ts [hunk 4]
```diff
@@ -49,7 +48,7 @@ export default class ReadDocument extends Controller {
     @tracked viewedPeriod: ViewPeriod = SEARCH_DEFAULT_VIEW_PERIOD;
     //workaround for bug w/array-based query param values
     //@see https://github.com/emberjs/ember.js/issues/18981
-    @tracked _searchTerms: string | null = JSON.stringify(SEARCH_DEFAULT_TERMS);
+    @tracked _searchTerms: string | null = null;
```
Comment (bakerac4) [inline pep/app/pods/read/document/controller.ts:51]: Do you need the comment above this anymore?

File: pep/app/pods/read/document/route.ts [hunk 5]
```diff
@@ -38,33 +43,57 @@ export default class ReadDocument extends PageNav(Route) {
...
-                pagingRootKey: null,
-                filterRootKey: null,
-                processQueryParams: (params) => ({ ...params, ...searchParams })
-            });
+            //if no search was submitted, don't fetch any results
+            if (hasSearchQuery(searchParams)) {
+                const controller = this.controllerFor(this.routeName);
+                const queryParams = buildQueryParams({
+                    context: controller,
+                    pagingRootKey: null,
```
Comment (bakerac4) [inline pep/app/pods/read/document/route.ts:83]: While it doesn't need to be fixed for this PR, I'd like to take a look at why we need to specify `pagingRootKey` and `filterRootKey` as null almost every time we use the pagination. We should just default it so we don't have to worry about it

Comment (billdami) [reply pep/app/pods/read/document/route.ts:83]: `buildQueryParams` is a util function that comes from our ember-pagination addon, which has its defaults optimized for JSON-API, which have the root keys for those as `page` and `filter`, which params get nested under. I guess we could maybe create an "overridden" version of this function in PEP that proxies to the addon's function w/those set to `null`? I would still keep the addon defaults to json-api tho is thats what most of our internal-api based projects will be using moving forward.

File: pep/app/pods/most-cited/template.hbs (hunk 2 above)
Comment (billdami) [reply pep/app/pods/most-cited/template.hbs:17]: The addon actually smartly adds an `aria-describedby` attribute on the anchor element (the `<button>` here) that points to the id of the tooltip content container, so we should be good accesibility-wise I think.

File: pep/app/pods/most-viewed/template.hbs (hunk 3 above)
Comment (billdami) [reply pep/app/pods/most-viewed/template.hbs:17]: see comment on above instance.

File: pep/app/pods/components/search/refine/component.ts (hunk 1 above)
Comment (billdami) [reply pep/app/pods/components/search/refine/component.ts:43]: yeah so its due to TS not knowing how ember's convenience `findBy()` returns, and that it is guaranteed to return an object here.

The alternative to the `!`, is the more verbose:
```
.map((id) => SEARCH_FACETS.find((facet) => facet.id === id));
```
But i'm fine with switching to that if you prefer (just losing the ability to use all of ember's nice convenience methods that we pay for in our vendor payload size is 😢)

**Pull request merged**